### PR TITLE
dev updates

### PR DESCRIPTION
**Pull request by claireh93**
Merge across latest changes from `main` into `dev`, including:

- Fixing Julia nightly runs through dropping JLD.
- Dropping JuliaDB support.
- Including new MPI testing.
- Fixing links from boydorr -> EcoJulia

### CODE REVIEW

**Comment by richardreeve** [comment]
@claireh93 - is this ready for me to check over?

**Comment by claireh93** [comment]
@richardreeve I think so!

**Review by richardreeve — APPROVED** [review]
I've made a few edits - there were a few changes to the examples to get them to run, and I finally fixed that warning about MPI when running their examples. The only problem is commenting out stuff in `src/ClimatePref/Plotting.jl`, but everything seems plausible to me. Obviously we'll have to do it again when the `claireh93/simple-sdm-layers` PR merges.

**Comment by richardreeve** [comment]
@claireh93 Now that #77 has merged, do you want to merge this and then do another PR to merge main in again? Or would you rather put it here - I'd suggest that doing a second PR might be easier!

**Comment by claireh93** [comment]
Yes I think I'll do a second PR, I think there's probably already plenty going on here! Thanks, I'm almost done looking through now

**Comment by claireh93** [comment]
Okay @richardreeve, I think this should be ready to merge once checks pass

File: src/Transitions/Run.jl [hunk 1]
```diff
@@ -341,12 +341,11 @@ end
 function simulate!(eco::AbstractEcosystem, times::Unitful.Time, timestep::Unitful.Time,
   cacheInterval::Unitful.Time, cacheFolder::String, scenario_name::String)
   time_seq = 0s:timestep:times
-  counting = 0
   for i in 1:length(time_seq)
       update!(eco, timestep, gettransitions(eco));
       # Save cache of abundances
       if mod(time_seq[i], cacheInterval) == 0year
-          JLD.save(joinpath(cacheFolder, scenario_name * (@sprintf "%02d.jld" uconvert(NoUnits,time_seq[i]/cacheInterval))), "abun", eco.abundances.matrix)
+          JLD2.@save joinpath(cacheFolder, scenario_name * (@sprintf "%02d.jld" uconvert(NoUnits,time_seq[i]/cacheInterval)))  abun=eco.abundances.matrix
```
Comment (richardreeve) [inline src/Transitions/Run.jl:348]: Does `@save` need a `JLD2` prefix?
```suggestion
          @save joinpath(cacheFolder, scenario_name * (@sprintf "%02d.jld" uconvert(NoUnits,time_seq[i]/cacheInterval)))  abun=eco.abundances.matrix
```

**Review by richardreeve — COMMENTED** [review]
Minor comment...

**Pull request merged**